### PR TITLE
mgr/dashboard: Remove username, password fields from Manager Modules/dashboard,influx

### PR DESCRIPTION
**Pull request by aaSharma14**
Username, password fields are empty in Cluster/Manager Modules/dashboard and Cluster/Manager Modules/influx. Since this functionality is when dashboard supported single user-password, now we need to remove these fields from here.

Fixes: https://tracker.ceph.com/issues/49645
Signed-off-by: Aashish Sharma <aasharma@redhat.com>

<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by nizamial09** [comment]
Test is failing on
```
____________ AccessControlTest.test_update_from_previous_version_v1 ____________

self = <dashboard.tests.test_access_control.AccessControlTest testMethod=test_update_from_previous_version_v1>

    def test_update_from_previous_version_v1(self):
        self.CONFIG_KEY_DICT['username'] = 'admin'
        self.CONFIG_KEY_DICT['password'] = \
            '$2b$12$sd0Az7mm3FaJl8kN3b/xwOuztaN0sWUwC1SJqjM4wcDw/s5cmGbLK'
        load_access_control_db()
>       user = self.exec_cmd('ac-user-show', username="admin")

tests/test_access_control.py:802: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/__init__.py:86: in exec_cmd
    return exec_dashboard_cmd(None, cmd, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

command_handler = None, cmd = 'ac-user-show', kwargs = {'username': 'admin'}
inbuf = None
cmd_dict = {'prefix': 'dashboard ac-user-show', 'username': 'admin'}, ret = -2
out = '', err = "User 'admin' does not exist"

    def exec_dashboard_cmd(command_handler, cmd, **kwargs):
        inbuf = kwargs['inbuf'] if 'inbuf' in kwargs else None
        cmd_dict = {'prefix': 'dashboard {}'.format(cmd)}
        cmd_dict.update(kwargs)
        if cmd_dict['prefix'] not in CLICommand.COMMANDS:
            ret, out, err = command_handler(cmd_dict)
            if ret < 0:
                raise CmdException(ret, err)
            try:
                return json.loads(out)
            except ValueError:
                return out
    
        ret, out, err = CLICommand.COMMANDS[cmd_dict['prefix']].call(mgr, cmd_dict, inbuf)
        if ret < 0:
>           raise CmdException(ret, err)
E           dashboard.tests.CmdException: User 'admin' does not exist

tests/__init__.py:51: CmdException
```

**Comment by aaSharma14** [comment]
jenkins test dashboard

**Review by avanthakkar — APPROVED** [review]
LGTM!

**Review by callithea — APPROVED** [review]

**Review by votdev — APPROVED** [review]
LGTM

When these module options are not used anymore, then we should consider to remove them from the MODULE_OPTIONS.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/mgr-modules/mgr-module-details/mgr-module-details.component.ts [hunk 1]
```diff
@@ -19,6 +19,8 @@ export class MgrModuleDetailsComponent implements OnChanges {
     if (this.selection) {
       this.mgrModuleService.getConfig(this.selection.name).subscribe((resp: any) => {
         this.module_config = resp;
+        delete this.module_config.username;
+        delete this.module_config.password;
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/mgr-modules/mgr-module-details/mgr-module-details.component.ts:23]: Wow, @aaSharma14 , what happened here? A couple of days ago it was a back-end refactoring and now it's just front-end hiding back-end stuff?

**Comment by aaSharma14** [comment]
jenkins test dashboard

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/mgr-modules/mgr-module-details/mgr-module-details.component.ts (hunk 1 above)
Comment (aaSharma14) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/mgr-modules/mgr-module-details/mgr-module-details.component.ts]: Back to back-end refactoring.

**Review by nizamial09 — APPROVED** [review]

**Review by avanthakkar — APPROVED** [review]

**Review by epuertat — APPROVED** [review]
Looks good to me @aaSharma14 ! I'm curious: as you're removing both v0 and v1 versions of the user db, have you checked which Ceph release last used v1? Probably nautilus already used v1, so perhaps it was mimic release?

**Comment by aaSharma14** [comment]
> Looks good to me @aaSharma14 ! I'm curious: as you're removing both v0 and v1 versions of the user db, have you checked which Ceph release last used v1? Probably nautilus already used v1, so perhaps it was mimic release?

nautilus release last used v1 https://github.com/ceph/ceph/blob/41b6adbf0045f27ec777eefc48c6ca3e2416e309/src/pybind/mgr/dashboard/services/access_control.py#L330

**Comment by votdev** [comment]
This PR causes a regression:

User databases with `version 1` created in Nautilus are not migrated automatically anymore to `version 2` which was introduced with Octopus

File: src/pybind/mgr/dashboard/services/access_control.py [hunk 2]
```diff
@@ -515,45 +515,6 @@ def accessdb_config_key(cls, version=None):
...
-                self.roles = {rn: Role.from_dict(r) for rn, r in v1_db.get('roles', {}).items()}
-                self.users = {un: User.from_dict(u, dict(self.roles, **SYSTEM_ROLES))
-                              for un, u in v1_db.get('users', {}).items()}
-
-                self.save()
-            else:
-                # If version 1 does not exist, check if migration of VERSION "0" needs to be done
-                check_migrate_v0_to_current()
-
-        check_migrate_v1_to_current()
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/services/access_control.py:555]: @votdev @p-se you're right, I think @aaSharma14 shouldn't have deleted the second method of this class...

Please feel free to revert that. We'll merge ASAP.

Comment (votdev) [reply src/pybind/mgr/dashboard/services/access_control.py:555]: I've created https://tracker.ceph.com/issues/51443 for that and working on it.

**Pull request merged**